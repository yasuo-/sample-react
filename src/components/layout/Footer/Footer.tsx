import React from 'react';

import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import { PATH } from '@/routes/path';

/**
 * Footer
 */
export const Footer: React.FC = () => {
  const { t } = useTranslation('common');
  const copyRightName = '....inc';
  const copayRightYear = new Date().getFullYear();

  const linkStyle = 'text-xs font-light tracking-tighter text-white/35 hover:text-white/100';
  return (
    <footer className="mx-auto mt-60 w-full bg-mine-shaft-800">
      <div className="flex justify-center justify-items-center py-2.5">
        <div>
          <Link to={PATH.TERMS} className={linkStyle}>
            {t('footer.terms')}
          </Link>
        </div>
        <div className="mx-2.5">
          <Link to={PATH.PRIVACY} className={linkStyle}>
            {t('footer.privacy')}
          </Link>
        </div>
        <div>
          <Link to={PATH.CONTACT} className={linkStyle}>
            {t('footer.contact')}
          </Link>
        </div>
      </div>
      <div className="flex justify-center justify-items-center pb-2.5">
        <span className="text-xs font-light tracking-tighter text-white/35">
          &copy;{copyRightName} {copayRightYear}
        </span>
      </div>
    </footer>
  );
};

Footer.displayName = 'Footer';
