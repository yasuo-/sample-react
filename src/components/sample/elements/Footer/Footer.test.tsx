import React from 'react';

import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Footer } from './Footer';

/**
 * renderWithRouter
 * helper function to render component with router
 * @param component
 */
const renderWithRouter = (component: React.ReactNode) => {
  return {
    ...render(<MemoryRouter>{component}</MemoryRouter>),
  };
};

describe('Footer Element', () => {
  it('displayName is Footer', () => {
    expect(Footer.displayName).toBe('Footer');
  });

  it('should take a snapshot', () => {
    const { asFragment } = renderWithRouter(<Footer />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders All rights reserved', () => {
    renderWithRouter(<Footer />);
    const screenElement = screen.getByText(/All rights reserved/i);
    expect(screenElement).toBeInTheDocument();
  });
});
