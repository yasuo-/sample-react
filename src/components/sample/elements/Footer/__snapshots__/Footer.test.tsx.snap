// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Element should take a snapshot 1`] = `[Function]`;
